coordinator/clients/traces-generator-api-client/src/main/kotlin/net/consensys/zkevm/coordinator/clients/TracesGeneratorJsonRpcClientV2.kt

```diff
@@ -65,10 +87,22 @@ class TracesGeneratorJsonRpcClientV2(
 
     return rpcClient.makeRequest(jsonRequest).toSafeFuture()
       .thenApply { responseResult ->
-        responseResult.mapEither(
+        val result = responseResult.mapEither(
           TracesClientResponsesParser::parseTracesCounterResponseV2,
           TracesClientResponsesParser::mapErrorResponseV2,
         )
+        if (config.ignoreTracesGeneratorErrors && !result.isSuccess()) {
+          Ok(createFallbackTracesCountersResponse())
+        } else {
+          result
+        }
+      }
+      .exceptionally { throwable ->
+        if (config.ignoreTracesGeneratorErrors) {
+          Ok(createFallbackTracesCountersResponse())
```

Comment:
Is this exceptionally branch really needed here? `if (config.ignoreTracesGeneratorErrors && !result.isSuccess())` should catch the errors from traces counter response, no?

Reply:
So what I'm confused is why would we want to return `createFallbackTracesCountersResponse` for network errors, I thought we only want to do so when we got bad responses from Tracer?

Reply:
It's only useful when `traces.conflation.request-retries.max-retries` are set. And if they are set and the Traces Generator goes down, conflation may just stop, because the max number of retries was reached and traces count or conflation process thrown an exception and it won't be retried

Reply:
It's only applicable to the cases when Dev Prover is used, I.e. we don't care about the finalization